component.go

```diff
@@ -25,136 +26,234 @@ type ChildLiveComponent interface{}
 
 //
 type LiveComponent struct {
-	Name               string
-	component          ComponentLifeTime
-	updatesChannel     *ComponentLifeCycle
-	htmlTemplateString string
-	htmlTemplate       *template.Template
-	rendered           string
-	IsMounted          bool
-	Prepared           bool
-	Exited             bool
-	log                Log
+	Name string
+
+	IsMounted bool
+	IsCreated bool
+	Exited    bool
+
+	log       Log
+	life      *ComponentLifeCycle
+	component ComponentLifeTime
+	renderer  LiveRenderer
+
+	children []*LiveComponent
 }
 
 // NewLiveComponent ...
 func NewLiveComponent(name string, time ComponentLifeTime) *LiveComponent {
 	return &LiveComponent{
 		Name:      name,
 		component: time,
+		renderer: LiveRenderer{
+			state:      &LiveState{},
+			template:   nil,
+			formatters: make([]func(t string) string, 0),
+		},
 	}
 }
 
-func (l *LiveComponent) getName() string {
-	return l.Name + "_" + NewLiveId().GenerateSmall()
-}
+func (l *LiveComponent) Create(life *ComponentLifeCycle) error {
+	var err error
 
-func (l *LiveComponent) RenderChild(fn reflect.Value, _ ...reflect.Value) template.HTML {
-	child, ok := fn.Interface().(*LiveComponent)
-	if !ok {
-		l.log(LogError, "child not a *golive.LiveComponent", nil)
+	l.life = life
 
-		return ""
-	}
+	// The first notification, will notify
+	// an component without unique name
+	l.notifyStage(WillCreate)
 
-	child.Mount()
+	l.Name = l.createUniqueName()
+
+	// Get the template defined on component
+	ts := l.component.TemplateHandler(l)
+
+	// Prepare the template content adding
+	// golive specific
+	ts = l.addGoLiveComponentIDAttribute(ts)
+
+	// Generate go std template
+	ct, err := l.generateTemplate(ts)
 
-	render, err := child.Render()
 	if err != nil {
-		l.log(LogError, "render child: render", logEx{"error": err})
+		return err
```

Comment:
You should wrap errors before returning e.g.

`return fmt.Errorf("generate template: %w", err)`